pkg/controllers/secretstore/common.go

```diff
@@ -156,3 +179,181 @@ func ShouldProcessStore(store esapi.GenericStore, class string) bool {
 
 	return false
 }
+
+// handleFinalizer manages the finalizer for ClusterSecretStores and SecretStores.
+func handleFinalizer(ctx context.Context, cl client.Client, store esapi.GenericStore, log logr.Logger, isPushSecretEnable bool) (finalizersUpdated bool, err error) {
+	// It adds a finalizer when there are PushSecrets with DeletionPolicy=Delete that reference this store
+	// and removes it when there are no such PushSecrets.
+	if !isPushSecretEnable {
```

Comment:
you only pass isPushSecretEnable for this check into handleFinalizer, so I think you should move isPushSecretEnabled outside of this function and then change the signature to 
```
func handleFinalizer(ctx context.Context, cl client.Client, store esapi.GenericStore, log logr.Logger) (finalizersUpdated bool, err error)
```

additionally you can receive a logger from the context so you can reduce to

```
func handleFinalizer(ctx context.Context, cl client.Client, store esapi.GenericStore) (finalizersUpdated bool, err error)
```